dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/dataintegrity/DefaultDataIntegrityService.java

```diff
@@ -84,6 +91,12 @@
 
     private I18nManager i18nManager;
 
+    private ProgramRuleService programRuleService;

```

Comment:
When enjecting your beans through the constructor, you can make these variables final as well.